src/main/java/az/subscriptionMS/service/impl/SubscriptionServiceImpl.java

```diff
@@ -0,0 +1,50 @@
+package az.subscriptionMS.service.impl;
+
+import az.subscriptionMS.model.Subscription;
+import az.subscriptionMS.repository.SubscriptionRepository;
+import az.subscriptionMS.service.SubscriptionsService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class SubscriptionServiceImpl implements SubscriptionsService {
+
+	@Autowired
+	private SubscriptionRepository subscriptionRepository;
+
+	@Override
+	public Subscription createSubscription(Subscription subscription) {
+		Subscription savedSubscription = subscriptionRepository.save(subscription);
+		//message
+		return savedSubscription;
+	}
+	@Override
+	@Cacheable(value = "Subscriptions", key = "#id")
+	public Optional<Subscription> getSubscription(Long id) {
+		return subscriptionRepository.findById(id);
+		//return subscriptionRepository.findById(id).orElseThrow(() -> new RuntimeException("Subscription not found !"));
+	}
+	@Override
+	public List<Subscription> getAllSubscriptions() {
+		return subscriptionRepository.findAll();
+	}
+	@Override
+	@CacheEvict(value = "subscriptions", key = "#id")
+	public Subscription updateSubscription(Subscription subscription) {
+		//message
+		return subscriptionRepository.saveAndFlush(subscription);
+	}
+	@Override
+	@CacheEvict(value = "subscriptions", key = "#id")
+	public void deleteSubscription(Long id) {
+		//message
+		subscriptionRepository.deleteById(id);
+	}
```

Comment:
dont use hard delete, use only soft delete. Change status from ACTIVE to INACTIVE